app/modules/pixbuf.h

```diff
@@ -0,0 +1,56 @@
+#ifndef APP_MODULES_PIXBUF_H_
+#define APP_MODULES_PIXBUF_H_
+
+enum pixbuf_type {
+  PIXBUF_TYPE_RGB /* 3-color: Green, Red, Blue order */,
+  PIXBUF_TYPE_GRB /* 3-color: Green, Red, Blue order */,
+  PIXBUF_TYPE_RGBW /* 4-color: RGB + White */,
+  PIXBUF_TYPE_GRBW /* 4-color: GRB + White */,
+  PIXBUF_TYPE_WWA /* 3-color: warm White, cold White, Amber */,
+
+  PIXBUF_TYPE_IBGR /* 3-color (Blue, Green, Red) & leading Intensity */,
+
+  PIXBUF_NTYPES
+};
+
+typedef struct pixbuf {
+  const size_t npix;
+  const uint8_t type;
+
+  /* FAM; true size is npix * pixbuf_channels_for(type) */
+  uint8_t values[0];
+} pixbuf;
+
+enum pixbuf_fade {
+  PIXBUF_FADE_IN,
+  PIXBUF_FADE_OUT
+};
+
+enum pixbuf_shift {
+  PIXBUF_SHIFT_LOGICAL,
+  PIXBUF_SHIFT_CIRCULAR
+};
+
+pixbuf *pixbuf_from_lua_arg(lua_State *, int);
+const size_t pixbuf_size(pixbuf *);
+
+// Exported for backwards compat with ws2812 module
+int pixbuf_new_lua(lua_State *);
+
+// Exported due to ws2812_effects; delete from here to HERE once that's gone
```

Comment:
Because if they're #ifdef'd out we won't get the compiler doing type-checking for us until someone builds with WS2812_EFFECTS.  It's harmless to have the prototypes in scope even without WS2812_EFFECTS, and we need the struct definition anyway.